core/src/main/java/org/apache/calcite/rel/metadata/RelMdExpressionLineage.java

```diff
@@ -186,7 +189,8 @@ protected RelMdExpressionLineage() {}
     // Extract input fields referenced by expression
     final ImmutableBitSet inputFieldsUsed = extractInputRefs(outputExpression);
 
-    if (rel.getJoinType() != JoinRelType.INNER) {
+    if (rel.getJoinType().generatesNullsOnLeft()
+        || rel.getJoinType().generatesNullsOnRight()) {
```

Comment:
Can we use `isOuterJoin` to do the check? Or just remove this check, add `FULL` check in `else` in L209.

Reply:
We can use `isOuterJoin` to do the check, thanks.